### PR TITLE
Ruby simple client

### PR DESCRIPTION
**Pull request by hectorcorrea**
A very simple client in Ruby to interact with MediaFlux. 

```
$ ruby mf.rb upload /acme my-file.txt
<response><reply type="result"><result><id>46180</id></result></reply></response>

$ ruby mf.rb get 46180
<response>...</response>

$ ruby mf.rb query "namespace='/acme'"
<response>...</response>
```

### CODE REVIEW

**Review by mccalluc — APPROVED** [review]
Cool -- It's useful to see these details. Before merging I'd suggest:
- Renaming the directory to something that describes this approach.
- Removing HTTP for safety.
- Adding references to the documentation that explain how the request encoding works... and if it's just generalizing from examples, that's good to know, but maybe add it to the list of things we ask our support person.

File: demos/mf.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+require "./mfclient.rb"
```
[the remark this answers is not in the document]
Comment (hectorcorrea) [reply demos/mf.rb:1]: Done in https://github.com/pulibrary/tiger-data/pull/24

File: demos/mfclient.rb [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+require "nokogiri"
+
+# A very simple client to interface with a MediaFlux server.
+class MediaFluxClient
+  def initialize(host, domain, user, password)
+    @host = host
+    @domain = domain
+    @user = user
+    @password = password
+    @base_url = "http://#{host}:80/__mflux_svc__/"
```
[the remark this answers is not in the document]
Comment (hectorcorrea) [reply demos/mfclient.rb:11]: Done in https://github.com/pulibrary/tiger-data/pull/24

File: demos/mfclient.rb [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    response_body = http_post(xml_request)
+  end
+
+  # Queries for assets on the given namespace
+  def query(aql_where)
+    xml_request = <<-XML_BODY
+      <request>
+        <service name="asset.query" session="#{@session_id}">
+          <args>
+            <where>#{aql_where}</where>
```
[the remark this answers is not in the document]
Comment (hectorcorrea) [reply demos/mfclient.rb:42]: Indeed.

File: demos/mfclient.rb [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      hex_bytes = []
+      # Force the string to be 16 characters long so we can guarantee 8 pairs.
+      number_hex = number.to_s(16).rjust(16, "0")
+      (0..7).each do |i|
+        n = i * 2
+        hex = number_hex[n..n+1]
+        hex_bytes << hex.to_i(16).chr
+      end
+      hex_bytes.join()
+    end
```
[the remark this answers is not in the document]
Comment (hectorcorrea) [reply demos/mfclient.rb:194]: I found this with WireShark...I am sure they are documented somewhere, but I have not found the documentation. I added a few comments in the code about this in https://github.com/pulibrary/tiger-data/pull/24

**Comment by hectorcorrea** [comment]
Moved this to https://github.com/pulibrary/tiger-data/pull/24

**Pull request closed without merging**